jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DefaultServletTest.java

```diff
@@ -813,6 +815,48 @@ public void testWelcomeMultipleBasesBase() throws Exception
         }
     }
 
+    @Test
+    public void testIncludedWelcomeDifferentBase() throws Exception
+    {
+        Path altRoot = workDir.getPath().resolve("altroot");
+        FS.ensureDirExists(altRoot);
+        Path altIndex = altRoot.resolve("index.html");
+
+        ServletHolder defholder = context.addServlet(DefaultServlet.class, "/alt/*");
+        defholder.setInitParameter("resourceBase", altRoot.toUri().toASCIIString());
+        defholder.setInitParameter("dirAllowed", "false");
+        defholder.setInitParameter("redirectWelcome", "false");
+        defholder.setInitParameter("welcomeServlets", "true");
+        defholder.setInitParameter("pathInfoOnly", "true");
+
+        ServletHolder gwholder = new ServletHolder("gateway", new HttpServlet()
+        {
+            @Override
+            protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+                    throws ServletException, IOException
+            {
+                req.getRequestDispatcher("/alt/").include(req, resp);
+            }
+        });
+        context.addServlet(gwholder, "/gateway/*");
```

Comment:
This addServlet should be done before Server.start()

Reply:
I don't think it's the case in other tests inside `org.eclipse.jetty.servlet.DefaultServletTest`...